src/RestServer/Controllers/WalletController.cs

```diff
@@ -0,0 +1,672 @@
+            var wallet = session.Wallet;
+            session.ResetExpiration();
+            if (wallet.VerifyPassword(model.OldPassword) == false)
+                throw new WalletException("Invalid password! Session terminated!");
+            if (model.CreateBackupFile && wallet is NEP6Wallet)
+            {
+                var bakFile = wallet.Path + $".{sessionId:n}.bak";
+                if (System.IO.File.Exists(wallet.Path) == false || System.IO.File.Exists(bakFile))
+                    throw new WalletException("Wallet backup failed.");
+                System.IO.File.Copy(wallet.Path, bakFile, model.OverwriteIfBackupFileExists);
+            }
+            if (wallet.ChangePassword(model.OldPassword, model.NewPassword) == false)
+                throw new WalletException("Failed to change password!");
+            if (wallet is NEP6Wallet nep6)
+                nep6.Save();
+            return Ok();
+        }
+
+        /// <summary>
+        /// Create a transaction with a script.
+        /// </summary>
+        /// <param name="sessionId" example="066843daf5ce45aba803587780998cdb">Session Id of the open/created wallet.</param>
+        /// <param name="model"></param>
+        /// <returns>Transaction object.</returns>
+        /// <response code="200">Successful</response>
+        /// <response code="400">If anything is invalid or request crashes.</response>
+        [HttpPost("{session:required}/transaction/script", Name = "WalletTransactionWithScript")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(Transaction))]
+        [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(ErrorModel))]
+        public IActionResult WalletTransactionWithScript(
+            [FromRoute(Name = "session")]
+            Guid sessionId,
+            [FromBody]
+            WalletTransactionScriptModel model)
+        {
+            if (WalletSessions.ContainsKey(sessionId) == false)
+                throw new KeyNotFoundException(sessionId.ToString("n"));
+            if (model.Script == null || model.Script.Length == 0)
+                throw new JsonPropertyNullOrEmptyException(nameof(model.Script));
+            if (model.From == null)
+                throw new JsonPropertyNullException(nameof(model.From));
+            var session = WalletSessions[sessionId];
+            session.ResetExpiration();
+            var wallet = session.Wallet;
+            var signers = model.Signers?.Select(s => new Signer() { Scopes = WitnessScope.CalledByEntry, Account = s }).ToArray();
+            var appEngine = ScriptHelper.InvokeScript(_settings, model.Script, signers);
+            if (appEngine.State != VM.VMState.HALT)
+                throw new ApplicationEngineException(appEngine.FaultException?.InnerException?.Message ?? appEngine.FaultException?.Message ?? string.Empty);
+            var tx = wallet.MakeTransaction(_neosystem.StoreView, model.Script, model.From, signers, maxGas: _settings.MaxInvokeGas);
```

Comment:
Can someone make sure that I'm doing this right? What I'm trying to do is have a script that is already built, than make a transaction of the built script. For example what's not working is that I have a script that just Invokes the Transfer function of the Neo token contract. like so _see below_ and its not popping the event. Also no Neo was transferred.
```csharp
ScriptHash.MakeScript("transfer", from, to, 1, "test")
```